### DISCUSSION TITLE
Including external IDF Component in build

### DISCUSSION THREAD

**Discussion by diogoviannaaraujo** [Moddable-OpenSource/moddable#1020] [Q&A]
Hi, I was trying to make a CBOR serializer and when looking at the examples on esp-idf repository it requires an external component. Seems to be defined in [this](https://github.com/espressif/esp-idf/blob/master/examples/protocols/cbor/main/idf_component.yml) file.

When I just try to include cbor.h I get that it could not be found

How can I make esp-idf include this cbor component in build?

**Comment by andycarle** [comment]
@diogoviannaaraujo You can generally list ESP-IDF component dependencies in the `CMakeLists.txt` file of the platform-appropriate build target's `main` project. For example, for the vanilla ESP32, the component dependencies are listed [here](https://github.com/Moddable-OpenSource/moddable/blob/819c387912fe0d8b53582b27c5739f1c261c3688/build/devices/esp32/xsProj-esp32/main/CMakeLists.txt#L39).

We do not currently have a mechanism for injecting those dependencies on a per-project basis. But you should be able to test by directly modifying that file in your Moddable SDK clone.

**Comment by diogoviannaaraujo** [comment]
That was fast! Thanks. Not really CI friendly but it will be enough for now.
